osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/DeviceRequestMessageProcessor.java

```diff
@@ -80,75 +87,93 @@ public void processMessage(final ObjectMessage message) throws JMSException {
         LOGGER.debug("Processing {} request message", this.messageType);
 
         MessageMetadata messageMetadata = null;
-        DlmsConnectionManager conn = null;
+        final DlmsConnectionManager conn = null;
         DlmsDevice device = null;
 
         try {
             messageMetadata = MessageMetadata.fromMessage(message);
 
-            /**
-             * The happy flow for addMeter requires that the dlmsDevice does not
-             * exist. Because the findDlmsDevice below throws a runtime
-             * exception, we skip this call in the addMeter flow. The
-             * AddMeterRequestMessageProcessor will throw the appropriate
-             * 'dlmsDevice already exists' error if the dlmsDevice does exists!
-             */
-            if (!MessageType.ADD_METER.name().equals(messageMetadata.getMessageType())) {
+            if (this.requiresExistingDevice()) {
                 device = this.domainHelperService.findDlmsDevice(messageMetadata);
             }
 
-            LOGGER.info("{} called for device: {} for organisation: {}", message.getJMSType(),
-                    messageMetadata.getDeviceIdentification(), messageMetadata.getOrganisationIdentification());
-
-            final Serializable response;
-            if (this.usesDeviceConnection()) {
-                conn = this.createConnectionForDevice(device, messageMetadata);
-                response = this.handleMessage(conn, device, message.getObject());
-            } else {
-                response = this.handleMessage(device, message.getObject());
-            }
+            LOGGER.info("{} called for device: {} for organisation: {}, correlationUID={}", message.getJMSType(),
+                    messageMetadata.getDeviceIdentification(), messageMetadata.getOrganisationIdentification(),
+                    messageMetadata.getCorrelationUid());
 
-            // Send response
-            this.sendResponseMessage(messageMetadata, ResponseMessageResultType.OK, null, this.responseMessageSender,
-                    response);
+            final RequestWithMetadata<S> request = this.createRequestWithMetadata(message, messageMetadata);
+            final Serializable response = this.getResponse(conn, device, request);
+            this.sendResponse(messageMetadata, response);
         } catch (final JMSException exception) {
             this.logJmsException(LOGGER, exception, messageMetadata);
         } catch (final Exception exception) {
-            // Return original request + exception
-            if (!(exception instanceof SilentException)) {
-                LOGGER.error("Unexpected exception during {}", this.messageType.name(), exception);
-            }
```

Comment:
I think the way this is refactored looks fine, but after the refactoring the `SuppressWarnings` annotation that was added in the past because of this check in an Exception catch block, no longer applies and should be removed from `processMessage`.

Reply:
removed SuppressWarnings annotation